test/test_pebble.py

```diff
@@ -520,53 +658,103 @@ def _services_action_async_helper(self, action, api_func, services):
         })
         change_id = api_func(timeout=0)
         self.assertEqual(change_id, '70')
-        self.assertEqual(self.api.requests, [
+        self.assertEqual(self.client.requests, [
             ('POST', '/v1/services', None, {'action': action, 'services': services}),
         ])
 
     def test_autostart_services(self):
-        self._services_action_helper('autostart', self.api.autostart_services, [])
+        self._services_action_helper('autostart', self.client.autostart_services, [])
 
     def test_autostart_services_async(self):
-        self._services_action_async_helper('autostart', self.api.autostart_services, [])
+        self._services_action_async_helper('autostart', self.client.autostart_services, [])
 
     def test_start_services(self):
         def api_func():
-            return self.api.start_services(['svc'])
+            return self.client.start_services(['svc'])
         self._services_action_helper('start', api_func, ['svc'])
 
     def test_start_services_async(self):
         def api_func(timeout=30):
-            return self.api.start_services(['svc'], timeout=timeout)
+            return self.client.start_services(['svc'], timeout=timeout)
         self._services_action_async_helper('start', api_func, ['svc'])
 
     def test_stop_services(self):
         def api_func():
-            return self.api.stop_services(['svc'])
+            return self.client.stop_services(['svc'])
         self._services_action_helper('stop', api_func, ['svc'])
 
     def test_stop_services_async(self):
         def api_func(timeout=30):
-            return self.api.stop_services(['svc'], timeout=timeout)
+            return self.client.stop_services(['svc'], timeout=timeout)
         self._services_action_async_helper('stop', api_func, ['svc'])
 
+    def test_service_error(self):
+        self.client.responses.append({
+            "change": "70",
+            "result": None,
+            "status": "Accepted",
+            "status-code": 202,
+            "type": "async"
+        })
+        change = self.build_mock_change_dict()
+        change['err'] = 'Some kind of service error'
+        self.client.responses.append({
+            "result": change,
+            "status": "OK",
+            "status-code": 200,
```

Comment:
Does pebble return a status: OK for an error response?

Reply:
Yeah, the overall request succeeded, even though one of the operations had an error. This makes a bit more sense when there are multiple changes/services at play -- for example, if you call `start('foo', 'bar')` and service `foo` starts but `bar` doesn't, overall it returns success even though one of the changes had an error.